File: checkers/cwe/tests/suite/NullPtrDereferenceChecker/java/conditional_expression_3.java
class A {
    String ret = "abc";
    String bar(bool flag) {
        return flag ? null : ret;
    }
    void foo() {
        String ret1 = bar(false);
        String ret2 = bar(true);
        ret1.length();   // no error here
        ret2.length();   // error here
    }
}

File: checkers/cwe/tests/suite/NullPtrDereferenceChecker/conditional_expression_3.cpp
#include <stdlib.h>

int* func(bool flag)
{
    static int ret = 1;
    return flag ? 0 : &ret;
}

int main()
{
    int* ret1 = func(false);
    int* ret2 = func(true);
    int a = *ret1;  // no error here
    return *ret2;   // error here
}
